internal/command/command.go

```diff
@@ -355,10 +356,246 @@ var CmdUpdateApis = &Command{
 			return nil
 		}
 
-		return push(ctx, languageRepo, startOfRun)
+		return push(ctx, languageRepo, startOfRun, "", "")
 	},
 }
 
+var CmdCreateReleasePR = &Command{
+	Name:  "create-release-pr",
+	Short: "Generate a PR for release",
+	Run: func(ctx context.Context) error {
+		err := checkFlags()
+		if err != nil {
+			return err
+		}
+		languageRepo, repoPath, inputDirectory, err := setupLanguageAndInputFolders(ctx)
+		if err != nil {
+			return err
+		}
+
+		pipelineState, err := loadState(languageRepo)
+		if err != nil {
+			fmt.Println("Error loading pipeline state:", err)
+			return err
+		}
+
+		librariesToRelease, prDescription, err := generateReleaseCommitForEachLibrary(ctx, repoPath, languageRepo, inputDirectory, pipelineState)
+		if err != nil {
+			return err
+		}
+
+		return generateReleasePr(ctx, librariesToRelease, pipelineState, repoPath, languageRepo, prDescription)
+	},
+}
+
+func setupLanguageAndInputFolders(ctx context.Context) (*gitrepo.Repo, string, string, error) {
+	startOfRun := time.Now()
+	tmpRoot, err := createTmpWorkingRoot(startOfRun)
+	var languageRepo *gitrepo.Repo
+	if flagRepoRoot == "" {
+		languageRepo, err = cloneLanguageRepo(ctx, flagLanguage, tmpRoot)
+		if err != nil {
+			return nil, "", "", err
+		}
+	} else {
+		languageRepo, err = gitrepo.Open(ctx, flagRepoRoot)
+		if err != nil {
+			return nil, "", "", err
+		}
+	}
+
+	inputDir := filepath.Join(tmpRoot, "inputs")
+	if err := os.Mkdir(inputDir, 0755); err != nil {
+		fmt.Println("Failed to create input directory:", err)
+		return nil, "", "", err
+	}
+
+	repoPath := flagRepoRoot
+	if repoPath == "" {
+		repoPath = filepath.Join(tmpRoot, fmt.Sprintf("google-cloud-%s", flagLanguage))
+	}
+
+	return languageRepo, repoPath, inputDir, nil
+}
+
+func checkFlags() error {
+	if !supportedLanguages[flagLanguage] {
+		return fmt.Errorf("invalid -language flag specified: %q", flagLanguage)
+	}
+	if flagImage == "" {
```

Comment:
We should use the pipeline state for the image - allow it to be overridden by a flag, but default to the one in the pipeline state. Basically `deriveImage` should already be okay for this, I think.